sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ModelsAsyncTest.java

```diff
@@ -0,0 +1,137 @@
+package com.azure.digitaltwins.core;
+
+import com.azure.core.http.HttpClient;
+import com.azure.digitaltwins.core.models.ModelData;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import reactor.test.StepVerifier;
+
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static com.azure.digitaltwins.core.TestHelper.DISPLAY_NAME_WITH_ARGUMENTS;
+import static com.azure.digitaltwins.core.TestHelper.assertRestException;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Async client implementation of the model tests defined in {@link ModelsTestBase}
+ */
+public class ModelsAsyncTest extends ModelsTestBase {
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.digitaltwins.core.TestHelper#getTestParameters")
+    @Override
+    public void modelLifecycleTest(HttpClient httpClient, DigitalTwinsServiceVersion serviceVersion) {
+        DigitalTwinsAsyncClient asyncClient = getAsyncClient(httpClient, serviceVersion);
+
+        // Create some models to test the lifecycle of
+        List<ModelData> createdModels = new ArrayList<>();
+        createModelsRunner(asyncClient, (modelsList) -> {
+            StepVerifier.create(asyncClient.createModels(modelsList))
+                .assertNext(createdModel -> {
+                    // When creating models, the service will not return the model itself, so it will be manually
```

Comment:
I am not sure I follow this - aren't the created models returned as a result of this API call - `PagedFlux<ModelData>`?

Reply:
Or is it that `ModelData.model` is null?

Reply:
ModelData.model is null when the service returns it from a createModel call. I'm reworking this code a bit